From 4260fbf479b684766ee1e21b74b37d556f043d38 Mon Sep 17 00:00:00 2001
From: Jason Perlow <jasonperlow@users.noreply.github.com>
Date: Wed, 17 Jun 2026 15:27:48 -0400
Subject: [PATCH 2016/2016] ALSA: hda/realtek: Add CIX Sky1 Phecda board fixup

Add ALC256_FIXUP_CIX_PHECDA (SSID 0x10ec:0x129e) which disables auto mic
switching and 3K pull-low so the internal DMIC stays usable with a headset
plugged in on the CIX Sky1 Phecda board. Hand-merged from the 6.18 LTS
fix onto the 7.0 alc269 fixup tables.
---
 sound/hda/codecs/realtek/alc269.c | 24 +++++++++++++++++++++++-
 1 file changed, 23 insertions(+), 1 deletion(-)

diff --git a/sound/hda/codecs/realtek/alc269.c b/sound/hda/codecs/realtek/alc269.c
index 04064ffd820c..797df622d6da 100644
--- a/sound/hda/codecs/realtek/alc269.c
+++ b/sound/hda/codecs/realtek/alc269.c
@@ -4133,6 +4133,7 @@ enum {
 	ALC245_FIXUP_CS35L41_I2C_2_MUTE_LED,
 	ALC236_FIXUP_HP_DMIC,
 	ALC245_FIXUP_HP_ENVY_X360_15_FH0XXX,
+	ALC256_FIXUP_CIX_PHECDA,
 };
 
 /* A special fixup for Lenovo C940 and Yoga Duet 7;
@@ -4169,6 +4170,22 @@ static void alc287_fixup_lenovo_yoga_book_9i(struct hda_codec *codec,
 	__snd_hda_apply_fixup(codec, id, action, 0);
 }
 
+static void alc_fixup_cix_phecda(struct hda_codec *codec,
+				 const struct hda_fixup *fix, int action)
+{
+	struct alc_spec *spec = codec->spec;
+
+	if (action == HDA_FIXUP_ACT_PRE_PROBE) {
+		/*
+		 * CIX Sky1 platform: disable auto mic switching by default
+		 * so users can use the internal DMIC even with headset plugged in.
+		 * Also disable 3K pull-low for headset detection.
+		 */
+		spec->en_3kpull_low = false;
+		spec->gen.suppress_auto_mic = true;
+	}
+}
+
 static const struct hda_fixup alc269_fixups[] = {
 	[ALC269_FIXUP_GPIO2] = {
 		.type = HDA_FIXUP_FUNC,
@@ -6685,7 +6702,11 @@ static const struct hda_fixup alc269_fixups[] = {
 		.v.func = cs35l41_fixup_i2c_two,
 		.chained = true,
 		.chain_id = ALC245_FIXUP_HP_X360_MUTE_LEDS
-	}
+	},
+	[ALC256_FIXUP_CIX_PHECDA] = {
+		.type = HDA_FIXUP_FUNC,
+		.v.func = alc_fixup_cix_phecda,
+	},
 };
 
 static const struct hda_quirk alc269_fixup_tbl[] = {
@@ -7431,6 +7452,7 @@ static const struct hda_quirk alc269_fixup_tbl[] = {
 	SND_PCI_QUIRK(0x10ec, 0x124c, "Intel Reference board", ALC295_FIXUP_CHROME_BOOK),
 	SND_PCI_QUIRK(0x10ec, 0x1252, "Intel Reference board", ALC295_FIXUP_CHROME_BOOK),
 	SND_PCI_QUIRK(0x10ec, 0x1254, "Intel Reference board", ALC295_FIXUP_CHROME_BOOK),
+	SND_PCI_QUIRK(0x10ec, 0x129e, "CIX Sky1 Phecda Board", ALC256_FIXUP_CIX_PHECDA),
 	SND_PCI_QUIRK(0x10ec, 0x12cc, "Intel Reference board", ALC295_FIXUP_CHROME_BOOK),
 	SND_PCI_QUIRK(0x10ec, 0x12f6, "Intel Reference board", ALC295_FIXUP_CHROME_BOOK),
 	SND_PCI_QUIRK(0x10f7, 0x8338, "Panasonic CF-SZ6", ALC269_FIXUP_ASPIRE_HEADSET_MIC),
-- 
2.51.0

